patroni/ha.py

```diff
@@ -921,11 +932,12 @@ def is_failover_possible(self, *, cluster_lsn: int = 0, exclude_failover_candida
         """
         candidates = self.get_failover_candidates(exclude_failover_candidate)
 
+        action = self._get_failover_action_name()
         if self.is_synchronous_mode() and self.cluster.failover and self.cluster.failover.candidate and not candidates:
-            logger.warning('Failover candidate=%s does not match with sync_standbys=%s',
-                           self.cluster.failover.candidate, self.cluster.sync.sync_standby)
+            logger.warning('%s candidate=%s does not match with sync_standbys=%s',
+                           action.title(), self.cluster.failover.candidate, self.cluster.sync.sync_standby)
         elif not candidates:
-            logger.warning('manual failover: candidates list is empty')
+            logger.warning('%s: candidates list is empty', action)
```

Comment:
Hmm, but this method is also called from `shutdown()`, and message could be misleading: `Manual failover: candidates list is empty`.

Reply:
https://github.com/zalando/patroni/pull/2784/files/ba204884d874b1805f48ada50578ae7255761c73#diff-b6a3a0ef64783a6d32feb908d74a3a009e9f25008b8ca5313c9c9747e3b74a63R217-R218 ("manual" can only appear inappropriately if there is an existing failover key during a shutdown/recover/demote/pg staring too long)